### DISCUSSION TITLE
How to automatically set syntax on after starting vscode?

### DISCUSSION THREAD

**Discussion by gaoqiangks** [vscode-neovim/vscode-neovim#1465] [General]
By default, `syntax off` is set by vscode-neovim. However, many plugins of neovim require `syntax on`. For the monent, I add two lines
`set updatetime=300` 
`au CursorHold * syntax on` 
to my init.vim to achieve this (I don't which event I should use. I tried many others and all failed) which is a bit awkward.  So, what's the correct way to do this?

**Comment by xiyaowong** [comment]
Try this
```vim
autocmd BufWinEnter * call timer_start(1000, { -> execute('setlocal syntax=on')})
```
By the way, which plugins that require 'syntax' do you use in VS Code?

**Reply by gaoqiangks** [reply]
I use vimtex. It extends the function of `%` so that I can jump between paired `$`. However, in vscode-neovim, it doesn't work. When `%` is pressed, the cursor always jumps to the previous `$` without `syntax on`.   I used to try some other plugins,  and failed with `syntax on`. I forget which plugins they are.

**Comment by xiyaowong** [comment]
vscode-neovim does not force syntax to be off now, it's enabled by default, and you can disable it by yourself.
